src/lib/Containers/Conversation.tsx

```diff
@@ -276,60 +288,85 @@ function sendConversationMessage(
   })
 }
 
-// TODO: Make this a pagination container instead of fetching 100 messages
-export default createRefetchContainer(
+// TODO Make this a pagination container instead of fetching 100 messages
+export default createPaginationContainer(
   Conversation,
-  graphql`
-    fragment Conversation_me on Me {
-      conversation(id: $conversationID) {
-        __id
-        id
-        from {
-          name
-          email
-          initials
-        }
-        to {
-          name
-          initials
-        }
-        messages(first: 200) @connection(key: "Conversation_messages") {
-          pageInfo {
-            hasNextPage
+  {
+    me: graphql.experimental`
+      fragment Conversation_me on Me {
+        conversation(id: $conversationID) @connection(key: "Conversation_messages") {
+          __id
+          id
+          from {
+            name
+            email
+            initials
           }
-          edges {
-            node {
-              impulse_id
-              is_from_user
-              body
-              ...Message_message
-            }
+          to {
+            name
+            initials
           }
-        }
-        items {
-          item {
-            ... on Artwork {
-              __typename
-              href
-              ...ArtworkPreview_artwork
+          messages(first: $count, after: $cursor) {
+            pageInfo {
+              startCursor
+              endCursor
+              hasPreviousPage
+              hasNextPage
+            }
+            edges {
+              cursor
+              node {
+                impulse_id
+                is_from_user
+                body
+                ...Message_message
+              }
             }
-            ... on Show {
-              __typename
-              href
-              ...ShowPreview_show
+          }
+          items {
+            item {
+              ... on Artwork {
+                __typename
+                href
+                ...ArtworkPreview_artwork
+              }
+              ... on Show {
+                __typename
+                href
+                ...ShowPreview_show
+              }
             }
           }
         }
       }
-    }
-  `,
-  graphql`
-    query ConversationRefetchQuery($conversationID: String!) {
-      me {
-        ...Conversation_me
+    `,
+  },
+  {
+    direction: "forward",
+    getConnectionFromProps(props) {
+      return props.me && props.me.conversation
+    },
+    getFragmentVariables(prevVars, totalCount) {
+      return {
+        ...prevVars,
+        count: totalCount,
       }
-    }
-  `
+    },
+    getVariables(props, { count, cursor }, fragmentVariables) {
+      return {
+        ...fragmentVariables,
+        count,
+        cursor,
+      }
+    },
+    query: graphql.experimental`
+      query ConversationPaginationQuery($conversationID: String!, $count: Int!, $cursor: String) {
+        me {
+          ...Conversation_me
```

Comment:
i think this also needs a `@arguments(count: $count, cursor: $cursor)`?

Reply:
Yes, but not in the query.